clang/test/CodeGenObjC/objc-asm-attribute-test.m

```diff
@@ -59,6 +63,10 @@ id Test16877359(void) {
 // CHECK: @"OBJC_CLASS_$_MySecretNamespace.Message" ={{.*}} global %struct._class_t
 // CHECK: @"OBJC_METACLASS_$_MySecretNamespace.Message" ={{.*}} global %struct._class_t
 
+// CHECK: @OBJC_PROP_NAME_ATTR_ = private unnamed_addr constant [13 x i8] c"optionalProp\00"
+// CHECK-NEXT: @OBJC_PROP_NAME_ATTR_.11 = private unnamed_addr constant [7 x i8] c"T@,?,&\00"
+// CHECK: @"_OBJC_$_PROP_LIST_MySecretNamespace.Protocol2" ={{.*}} [%struct._prop_t { ptr @OBJC_PROP_NAME_ATTR_, ptr @OBJC_PROP_NAME_ATTR_.11 }]
```

Comment:
For the 5.10 branch, shouldn't this be a `i8* getelementptr` instead of `ptr`?

This is what I seem to get when I execute the test (newlines added):

```
@"_OBJC_$_PROP_LIST_MySecretNamespace.Protocol2" = internal global {
i32, i32, [1 x %struct._prop_t] } { i32 16, i32 1, [1 x %struct._prop_t]
[%struct._prop_t { i8* getelementptr inbounds ([13 x i8], [13 x i8]* @OBJC_PROP_NAME_ATTR_, i32 0, i32 0),
i8* getelementptr inbounds ([7 x i8], [7 x i8]* @OBJC_PROP_NAME_ATTR_.11, i32 0, i32 0) }]
}, section "__DATA, __objc_const", align 8
```

Reply:
It passed PR testing, so at the time that happened it must have been right, surely? If you're right then we probably need to tweak the test a little. Which platform are you seeing that on?

Reply:
Oh no :-( It occurs to me that the PR testing is all x86_64. I wonder if this is an ARM64 thing?

Reply:
Both Linux and Darwin, but both in x86_64.

As far as I understand PR testing does not run check-llvm nor check-clang, which leaves so broken tests go through sometimes.

Reply:
I'll update the test and raise a PR (I'll need to go and build the branch first, just so I can check that I've actually fixed it).

Reply:
PR is https://github.com/apple/llvm-project/pull/7625